pinot-common/src/main/java/org/apache/pinot/common/request/context/ExpressionContext.java

```diff
@@ -117,7 +117,7 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    int hash = 31 * 31 * _type.hashCode();
+    int hash = 31 * 31 * _type.toString().hashCode();
```

Comment:
Why? It is adding unnecessary overhead

Reply:
true, however for enum, the hashcode() method is stable within the JVM process, but not stable across multiple JVMs. to ensure the hashcode is stable, we need to use the string hash.

Maybe we could cache all the hashcode in a Map to speed up.